### PR TITLE
update to 3.15.0

### PR DESCRIPTION
**Pull request by katietz**

### CODE REVIEW

**Comment by katietz** [comment]
There is no breaking change mention in Changelogs on https://www.pycryptodome.org
Tests using nowflake-connector-python showed no issue

File: recipe/meta.yaml [hunk 1]
```diff
@@ -57,7 +59,7 @@ test:
     - psutil
 
 about:
-  home: http://www.pycryptodome.org
+  home: https://www.pycryptodome.org
   license: Public Domain/BSD 2-Clause/OCB/MPIR
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:63]: The licenses should be: `Unlicense AND BSD-2-Clause`, see https://github.com/Legrandin/pycryptodome/blob/v3.15.0/LICENSE.rst

**Review by cbouss — APPROVED** [review]
LGTM

**Review by varlackc — COMMENTED** [review]
We might want to add the `License_family`
https://github.com/AnacondaRecipes/pycryptodomex-feedstock/blob/12904bd0194ecffa2c2cb900e2d02c737fb17714/recipe/meta.yaml#L62&L65

File: recipe/meta.yaml (hunk 1 above)
Comment (katietz) [reply recipe/meta.yaml:63]: well, actually Unlicense is PD ... I consider them as equivalent

**Comment by katietz** [comment]
so what the license_family should be? PD? BSD? or OTHER?   I think we can skip this item here

**Pull request merged**